telegram/ext/_application.py

```diff
@@ -1672,3 +1688,64 @@ async def process_error(
 
         _logger.exception("No error handlers are registered, logging exception.", exc_info=error)
         return False
+
+
+class BaseProcessor:
+    def __init__(self, max_concurrent_updates: int):
+        self.max_concurrent_updates = max_concurrent_updates
+        self.semaphore = BoundedSemaphore(self.max_concurrent_updates or 1)
```

Comment:
```suggestion
        self._semaphore = BoundedSemaphore(self.max_concurrent_updates or 1)
```

Do you also want to check for negative values here?

Reply:
Yeah, Gonna add that